src/trex/molecule.py

```diff
@@ -82,3 +85,88 @@ def compute_consensus(sequences):
     bin_consens = np.argmax(matrix, axis=1)
     # convert maximum counts into consensus sequence
     return "".join(letters[bin_consens])
+
+
+def remove_too_short_clone_ids(molecules: List[Molecule], min_length: int) -> List[Molecule]:
+    """
+    Discard cloneIDs shorter than min_length
+    """
+    return [
+        m for m in molecules
+        if len(m.clone_id.replace("0", "").replace("-", "")) >= min_length
+    ]
+
+
+def correct_clone_ids_per_cell(
+    molecules: List[Molecule], max_hamming: int, min_overlap: int = 20
+) -> List[Molecule]:
+    """
+    Attempt to correct sequencing errors in the CloneID sequences of all
+    molecules looking for similar sequences in the same cell.
+    """
+    # Obtain all cloneIDs (including those with '-' and '0')
+    clone_ids = [m.clone_id for m in molecules]
+
+    # Count the full-length cloneIDs
+    counts = Counter(clone_ids)
+
+    cell_dict = defaultdict(list)
+    for molecule in molecules:
+        cell_dict[molecule.cell_id].append((molecule.umi, molecule.clone_id))
```

Comment:
You can just append the full molecule:
```suggestion
        cell_dict[molecule.cell_id].append(molecule)
```